### DISCUSSION TITLE
Ring API auth

### DISCUSSION THREAD

**Discussion by SCMicroMike** [wictorwilen/ring-timelapse#4] [General]
Sorry I am very new to coding, Can you please assist me with how to get  the Ring Token. I am seeing an error that npx is not recognized.

**Comment by wictorwilen** [comment]
@SCMicroMike - verify that you have *node* and *npx* installed (https://www.npmjs.com/package/npx)

**Reply by SCMicroMike** [reply]
Thank you for the response, I am sorry if this is too basic a question. Do I need to install ring-client-api and all its prerequisites in this Docker image or are these already in the image. I think I am following, but I am so new to this all. Trying to get Youtube smart on this stuff.

**Reply by wictorwilen** [reply]
All of that should be in the Docker image. All you need to do is to get that authorization token and pass that as an argument to the Docker image

**Comment by wabragg** [comment]
Yeah, this is the most confusing part of this Docker package ... the need for 'npx'.  Never heard of it.  It might help to expand on the documentation a bit to help people understand that they will need to download and install the npm tools.  And I assume by 'node' you are referring to 'nodejs'.  There would probably be a lot more pulls of this container if these dependencies were more clear.  Most people are probably at the CLI with root access typing 'npx', getting the obligatory 'command not found' and moving on.
